net/unbound/files/unbound.init

```diff
@@ -1,20 +1,718 @@
+    echo "  control-interface: ::1" >> $UNBOUND_CONF
+    
+  else  
+    echo "remote-control:" >> $UNBOUND_CONF
+    echo "  control-enable: no" >> $UNBOUND_CONF
+  fi
+}  
+
+
+##############################################################################
+
+update_root_files() {
+  drillpath=$( which drill )
+  digpath=$( which dig )
+  anchorpath=$( which unbound-anchor )
+  
+  if [ -x "$drillpath" ] ; then 
+    drillpath="$drillpath -D"
+    
+  elif [ -x "$digpath" ] ; then
+    drillpath="$digpath +dnssec"
+    
+  else 
+    logger -t unbound -s "need drill, dig, or something"
+    return 0
+  fi
+
+  
+  # Built in unbound-update is way too busy; burn out flash ROM.
+  # Only does DNSKEY but root servers rotate more ... what?
+  if [ -f /etc/unbound/root.hints ] ; then
+    root_date=$( date -r /etc/unbound/root.hints +%s )
+    now_date=$( date +%s )
+    delta_days=$(( (now_date - root_date) / 86440 ))
+    logger -t unbound -s "root domain is $delta_days days old"
+    
+  else
+    logger -t unbound -s "no root hints file to start with"
+    return 0
+  fi
+  
+
+  if [ "$delta_days" -gt "$UNBOUND_N_ROOT_AGE" ] ; then
+    # We need to bootstrap unbound, it aint runn'n yet.
+    rootservers=$( awk '{ if ($4 == "A") { print $5 } }' /etc/unbound/root.hints )
+    
+    for rootserver in $rootservers ; do
+      $drillpath NS . @$rootserver > /tmp/root.hints.new
+      filestuff=$( awk '{ if ($4 == "A") { print $5 } }' /tmp/root.hints.new )
+      
+      
+      case "$filestuff" in
+        [1-9]*"."[0-9]*"."[0-9]*"."[1-9]*)
+          logger -t unbound -s "root.hints updated @$rootserver"
+          cp /tmp/root.hints.new /etc/unbound/root.hints
+          rm /tmp/root.hints.new
+          break
+          ;;
+
+        *)
+          logger -t unbound -s "root.hints not found @$rootserver" 
+          ;;
+      esac
+    done
+    
+    
+    if [ -x "$anchorpath" ] ; then
+      # Use unbound-anchor optional helper
+      $anchorpath -a /etc/unbound/root.key -r /etc/unbound/root.hints
+    
+    else
+      # Like unbound-anchor default fast method, but use drill
+      $drillpath DNSKEY . @$rootserver > /tmp/root.key.new
+      filestuff=$( cat /tmp/root.key.new )
+      
+      
+      case "$filestuff" in
+        *NOERROR*)
+          logger -t unbound -s "root.key updated @$rootserver"
+          cp /tmp/root.key.new /etc/unbound/root.key
+          rm /tmp/root.key.new 
+          ;;
+
+        *) 
+          logger -t unbound -s "root.key failed update @$rootserver" 
+          ;;
+      esac
+    fi
+  fi
+}
+
+##############################################################################
+
+start_service () {
+  mkdir -p $( dirname $UNBOUND_CONF )
+  echo > $UNBOUND_CONF
+    
+  procd_open_instance
+  procd_set_param command $PROG -d -c $UNBOUND_CONF
+  procd_set_param respawn
+  procd_close_instance
+  
+  config_load unbound
+  config_foreach build_config_file unbound
+  update_root_files
```

Comment:
**Update:** confirmed with Unbound team that root.hints is updated every time Unbound starts, root zone TTL expires (86400, 1d), or whenever a hick-up requires fetching the DNSKEY again. This is really hard on flash ROM.